### PR TITLE
(GH-4) Added ability to load the embedded warp files during addin load

### LINKED ISSUE #4: Embedd the warp binaries and extract the correct binary when the library gets loaded
This is possible to do by using one of the Fody Weavers, and would allow the addin to not have any external dependencies (Which would be the best as there is currently no installation for Warp, only download and execute(

### PR DESCRIPTION
**Pull request by AdmiringWorm**
## Description

The loading of the warp files is done by using Fody IL Weaving using the weaver called ModuleInit.
It will call the class named `ModuleInitializer`, I also make use of another weaver called Resourcer which have the responsibility to make it easier to get a stream to the embedded file (it also makes sure that the embedded file exist during compilation).

Additionally there is also an internal configuration file, which will store the path to the extracted warp executable. (That path is planned to be used in the Addin runner).

## Checklist

- [x] All tests are passing
- [x] New tests were created to address changes in pr (and tests are passing)
- [ ] Updated README and/or documentation, if necessary
- [x] Added myself / the `contributors` by using the allcontributors bot (or will do)

Thanks for contributing!

fixes #4

### CODE REVIEW

File: src/Cake.Warp/ModuleInitializer.cs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        /// the loading of the addin library (by Module.Fody IL weaving).
+        /// </summary>
+        /// <notes>
+        ///   While this method is public, it should never be
+        ///   called manually. It should only be called
+        ///   using the existing IL weaving.
+        /// </notes>
+        public static void Initialize()
+        {
+            // TODO
```
Comment (gep13) [inline src/Cake.Warp/ModuleInitializer.cs]: What is there left to do here?

Comment (AdmiringWorm) [reply src/Cake.Warp/ModuleInitializer.cs]: I completely forgot to add context to that todo statement, there is still something that needs to be done though.
I need to add an early check if the file exist, in which case we can skip the extraction of the binary file.

Another thing I forgot, is to investigate if the executing bit needs to be set for the binary extracted on linux and OSX. (I do believe that would be needed)

**Comment by AdmiringWorm** [comment]
I would also appreciate it, if you don't see anything that don't need to change (or nothing that you recommend to be changed), that you give an approved review.
This way, I can test the settings I have for mergify

**Comment by gep13** [comment]
@AdmiringWorm apologies, just getting to a point to look at this.  Been a busy weekend!

Had a closer look at this, and I don't see an issue with any of the code that you are using.  Seems good to me!

**Pull request closed without merging**